sdk/iot/core/inc/az_iot_hub_client.h

```diff
@@ -55,7 +55,7 @@ AZ_NODISCARD az_iot_hub_client_options az_iot_hub_client_options_default();
 /**
  * @brief Initializes an Azure IoT Hub Client.
  *
- * @param[out] client
+ * @param[out] client The #az_iot_hub_client to use for this call

```

Comment:
The build system flags it as an undocumented param which is why I had to go in and add it to all of the calls. Won't pass the gate otherwise. And I don't see a reason why we shouldn't have at least the most mundane documentation for everything. In this case, it gives the user easy access to check the documentation for the `az_iot_hub_client` struct by providing a link right then and there if they want it. Otherwise, they have to do additional hunting or searching for the docs for this struct.

Reply:
Makes sense: could you please update the provisioning one to ensure we use the same pattern?
Also: I think the other docs end in ".". Please make sure we're consistent.

Reply:
I will after this PR just to keep scopes.